README.md

```diff
@@ -268,83 +259,47 @@ The steps below summarise the possible steps in defining such a convention:
 
 * declare a new convention,
 * add explicit `scitype` (and `Scitype`) definitions,
-* register any traits that were needed to define scitypes,
 * optionally define `coerce` methods for your convention
 
 Each step is explained below, taking the MLJ convention as an example.
 
 ### Naming the convention
 
-In the module, define a
+In the module, define a singleton as thus
 
 ```julia
 struct MyConvention <: ScientificTypesBase.Convention end
 ```
 
-and add an init function with:
-
-```julia
-function __init__()
-  ScientificTypesBase.set_convention(MyConvention())
-end
-```
-
 ### Adding explicit `scitype` declarations.
 
 When overloading `scitype` one needs to dipatch over the convention,
 as in this example:
 
 ```julia
-ScientificTypesBase.scitype(::Integer, ::MLJ) = Count
+ScientificTypesBase.scitype(::Integer, ::DefaultConvention) = Count
 ```
-
-In some cases, however, the scientific type to be attributed to an
-object might depend on the evaluation of a boolean-valued trait
-function. There is a mechanism for "registering" such traits to
-streamline trait-based dispatch of the `scitype` method. This is best
-illustrated with an example.
-
-In the MLJ convention, all containers that meet the
-[`Tables.jl`](https://github.com/JuliaData/Tables.jl) interface are
-deemed to have scitype `Table`. These are detected using the Tables.jl
-trait `istable`. Our first step is to choose a name for the trait, in
-	this case `:table`. Our `scitype` declaration then reads:
-
-```
-function ScientificTypesBase.scitype(X, ::MLJ, ::Val{:table})
-   K = <some type depending on columns of X>
-   return Table{K}
-end
+To avoid method ambiguities, avoid dispatching only on the first argument.
+For example, defining
+```julia
+ScientificTypesBase.scitype(::AbstractFloat, C) = Continous
 ```
-
-For this to work we now need to register the trait, which means adding
-to the `TRAIT_FUNCTION_GIVEN_NAME` dictionary, which should be
-performed within the init function of the defining package:
-
+would lead to ambiguities in another package defining 
 ```julia
-function __init__()
-    ScientificTypesBase.set_convention(MLJ())
-    ScientificTypesBase.TRAIT_FUNCTION_GIVEN_NAME[:table] = Tables.istable
-end
+ScientificTypesBase.scitype(a, ::DefaultConvention) = Count
```

Comment:
Shouldn't we also add something like:

Since `ScientificTypesBase.jl` does not define a single-argument `scitype(X)` method, a new implementation of a scientific convention will typically want also to explicitly implement this method, to save her users from needing to explicitly specify a convention. So, for example, one declares:

```julia
ScientificTypesBase.scitype(X) = scitype(X, MyConvention())
```

?